modules/@apostrophecms/piece-type/index.js

```diff
@@ -1060,6 +1088,105 @@ module.exports = {
+          }
+
+          console.log('Adding drafts and locales to documents');
+
+          const locales = Object.keys(self.apos.i18n.locales);
+          const lastPublishedAt = new Date();
+          let deletes = [];
+          let inserts = [];
+
+          await self.apos.migration.eachDoc({ type: self.name }, async doc => {
+            if (doc.aposDocId && !doc._id.endsWith('published') && !doc._id.endsWith('draft')) {
+              deletes.push(doc._id);
+
+              for (const locale of locales) {
+                const newDraft = {
+                  ...doc,
+                  aposLocale: `${locale}:draft`,
+                  aposMode: 'draft',
+                  aposDocId: doc._id,
+                  _id: `${doc.aposDocId}:${locale}:draft`
+                };
+                const newPublished = {
+                  ...doc,
+                  aposLocale: `${locale}:published`,
+                  aposMode: 'published',
+                  aposDocId: doc._id,
+                  _id: `${doc.aposDocId}:${locale}:published`,
+                  lastPublishedAt
+                };
+                inserts.push(newDraft);
+                inserts.push(newPublished);
+
+                if (inserts.length > 100) {
+                  self.flush(deletes, inserts);
+                  deletes = [];
+                  inserts = [];
+                }
+              }
+            }
+          });
+
+          self.flush(deletes, inserts);
+          await self.apos.attachment.recomputeAllDocReferences();
+
+          console.log(`Done localizing module ${self.name}`);
+        }
+      },
+
+      unlocalize: {
+        usage: 'Remove duplicate documents when a module has not "localized" and "autopublish" anymore.' +
+        '\nOptions are:' +
+        '\n- locale: if not set, it is the project\'s default locale' +
+        '\n- mode: by default, published' +
+        '\nExample: node app [moduleName]:unlocalize --mode=published --locale=en',
+        async task(argv) {
+          if (self.options.localized) {
+            throw new Error('Localized option not set to false, so the module cannot be unlocalized.');
+          }
+
+          const locale = argv.locale || self.apos.i18n.defaultLocale;
+          const mode = argv.mode || 'published';
+          let deletes = [];
+          let inserts = [];
+
+          console.log(`Removing duplicated documents and updating ${mode} ones`);
+
+          await self.apos.migration.eachDoc({ type: self.name }, async doc => {
+            deletes.push(doc._id);
+
+            if (doc.aposDocId && doc.aposLocale === `${locale}:${mode}` && doc.aposMode === mode) {
+              const newDoc = {
+                ...doc,
+                aposLocale: undefined,
+                aposMode: undefined,
+                _id: doc.aposDocId
+              };
+              inserts.push(newDoc);
+
+              if (inserts.length > 100) {
```

Comment:
This is great, but how about moving this check into `self.flush` itself, along with the cleanup of the two arrays. This also makes it easier to check the length of both "inserts" and "deletes".

To clean them up even though they are passed in as arguments, take advantage of "splice".

The less code you have to write every time you call flush, the less can go wrong.

Also I suggest renaming it "flushInsertsAndDeletes" just to avoid camping on a potentially common method name.